### ISSUE TITLE
Flatpak version crashes when saving settings

### ISSUE THREAD

**Issue by Parnassius** [exelix11/SysDVR#255]
**Describe the bug**
Flatpak version crashes when saving settings

**To Reproduce**
- Open the Flatpak version of SysDVR-Client
- Click the Settings button
- Click the Save changes button

**Expected behavior**
Settings should get saved

**Setup information**
 - PC Os: openSUSE Tumbleweed (Linux x64)
 - Console firmware: N/A
 - Custom firmware name and version: N/A
 - SysDVR version: N/A

**Additional context**
This is the error log that shows up as soon as I try to save the settings:
```
System.IO.IOException: Read-only file system : '/app/bin/options.json'
   at Interop.ThrowExceptionForIoErrno(Interop.ErrorInfo, String, Boolean) + 0x19
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, Interop.Sys.OpenFlags, Int32, Boolean, Boolean&, Func`4) + 0x13f
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String, FileMode, FileAccess, FileShare, FileOptions, Int64, UnixFileMode, Int64&, UnixFileMode&, Boolean, Boolean&, Func`4) + 0x11c
   at System.IO.File.OpenHandle(String, FileMode, FileAccess, FileShare, FileOptions, Int64) + 0x96
   at System.IO.File.WriteToFile(String, FileMode, String, Encoding) + 0x81
   at SysDVR.Client.GUI.OptionsView.Draw() + 0x124
   at SysDVR.Client.ClientApp.EntryPoint() + 0x2e2
   at SysDVR.Client.Program.RunApp(String[]) + 0x2b8
```

The issue seems to originate from the value of [`Resources.SettingsStorePath`](https://github.com/exelix11/SysDVR/blob/v6.0/Client/Platform/Resources.cs#L92), which is set to the folder that contains the SysDVR-Client executable. On Flatpak this is `/app/bin`, which is not user-writable.

A possible fix could be setting that variable to a subfolder of `$XDG_CONFIG_HOME` (usually `~/.var/app/com.github.exelix11.sysdvr/config/` when running in a Flatpak), adding a check in [`SystemUtil.StoreSettingsString`](https://github.com/exelix11/SysDVR/blob/v6.0/Client/Platform/SystemUtil.cs#L61-L65) to make sure the folder exists before saving.

Sadly I know basically nothing about dotnet, so I'm unable to create a pull request to fix this.

Thank you

**Comment by exelix11** [comment]
It's fine, thanks for providing the insight needed to fix the issue so i can get to work on it right away.

**Comment by exelix11** [comment]
The issue seems fixed on my end, can you try latest flatpak from github actions https://github.com/exelix11/SysDVR/actions/runs/7426921602/artifacts/1150957073 ?

**Comment by Parnassius** [comment]
The latest commit fixed the issue for me as well, thank you!

**Comment by exelix11** [comment]
I also updated the release, if anyone else is affected just redownload the 6.0 flatpak. The fixed version shows build id `e18a704`
